### ISSUE TITLE
Don't assume I want to add a weapon to a comparison just because I click it with one already open

### ISSUE THREAD

**Issue by chaosblad3** [DestinyItemManager/DIM#7887]
### Proposed change

While using the item comparison feature, clicking on another weapon should NOT automatically try to add it to the current comparison list, or throw up an error message if the weapon type of the item clicked does not match the item currently being compared.

Instead it should bring up the menu and details for that item like normal (ie when no comparison is already open), but instead of the normal "Compare" button there should be two, one marked "Compare (add)" which attempts to add the item to the current comparison if the weapon type matches, and one marked "Compare (new)" which automatically clears the current comparison and starts a new one from the selected item.

Alternatively, add a toggle option to the comparison viewer itself that controls whether clicking another item tries to add it or just opens it's menu like normal.

But most importantly it should be possible to perform other operations (lock/unlock, transfer, swap active perks/shaders/etc) with the selected item without interfering with a current comparison.

### How does this fit into your workflow?

Often times I will be in the process of comparing a particular weapon while trying to farm for a particular roll, and then for whatever reason I will need to click on an item of a different weapon type, e.g. to check a roll or transfer an item from the postmaster etc, but don't want to have to close the current comparison first, perform the operation, and then have to reopen the previous comparison again afterwards if I still need it, and getting an error message saying "This item doesn't match the type of items being compared." when I _wasn't actually **trying** to compare it_, and then thusly not being able to perform the operation I actually intended without first closing the comparison is profoundly infuriating.

Basically I do not want you to automatically assume I am trying to compare an item just because I click on an item with a comparison currently open, if I actually wanted to add the weapon I clicked to the current comparison then my preposed method would only take one extra click per item (or one extra click total with the toggle method), but compared to having to close the current comparison before I am allowed to interact with any other weapon type and then if I still need that comparison, having to reopen it again afterwards is significantly more counterproductive.

**Comment by chaosblad3** [comment]
Sample mock-up for possible toggle option:
![image](https://user-images.githubusercontent.com/4968222/154482058-8a410876-f14e-40f3-ba2e-a24313381e55.png)

**Comment by bhollis** [comment]
Compare is central enough to so many people's workflow that I'd rather not change how it behaves. While we of course mechanically know how to add a toggle for this behavior, we generally avoid such a thing.
